source/non_matching/coupling.cc

```diff
@@ -285,26 +285,105 @@ namespace NonMatching
+    const unsigned int n_active_c =
+      immersed_dh.get_triangulation().n_active_cells();
+    std::vector<Point<spacedim>> all_points(n_active_c * n_q_points);
+
+    // Collecting all points
+    {
+      unsigned int c = 0;
+      for (const auto &cell : immersed_dh.active_cell_iterators())
+        {
+          // Reinitialize the cell and the fe_values
+          fe_v.reinit(cell);
+          const std::vector<Point<spacedim>> &x_points =
+            fe_v.get_quadrature_points();
+
+          // Copy the points to the vector
+          std::copy(x_points.begin(),
+                    x_points.end(),
+                    all_points.begin() + c * n_q_points);
+          ++c;
+        }
+    }
+
+    // Get a list of outer cells, qpoints and maps.
+    const auto  cpm = GridTools::compute_point_locations(cache, all_points);
+    const auto &all_cells   = std::get<0>(cpm);
+    const auto &all_qpoints = std::get<1>(cpm);
+    const auto &all_maps    = std::get<2>(cpm);
+
+    std::vector<
+      std::vector<typename Triangulation<dim0, spacedim>::active_cell_iterator>>
+                                                       cell_container(n_active_c);
+    std::vector<std::vector<std::vector<Point<dim0>>>> qpoints_container(
+      n_active_c);
+    std::vector<std::vector<std::vector<unsigned int>>> maps_container(
+      n_active_c);
+
+    // Cycle over all cells of underling mesh found
+    // call it omesh, elaborating the output
+    for (unsigned int o = 0; o < all_cells.size(); ++o)
+      {
+        for (unsigned int j = 0; j < all_maps[o].size(); ++j)
+          {
+            // Find the index of the "owner" cell and qpoint
+            // w.r.o. the immersed mesh
+            unsigned int cell_id = all_maps[o][j] / n_q_points;
+            unsigned int n_pt    = all_maps[o][j] % n_q_points;
+
+            // Checking if the cell was already added
+            // first we look at the last inserted cell, which is more likely:
+            if (!cell_container[cell_id].empty() &&
+                cell_container[cell_id].back() == all_cells[o])
+              {
+                qpoints_container[cell_id].back().push_back(all_qpoints[o][j]);
+                maps_container[cell_id].back().push_back(n_pt);
+              }
+            else
+              {
+                auto cell_p = std::find(cell_container[cell_id].begin(),
+                                        cell_container[cell_id].end(),
+                                        all_cells[o]);
+                if (cell_p == cell_container[cell_id].end())
+                  {
+                    const std::vector<Point<dim0>> new_qpoint = {
+                      all_qpoints[o][j]};
+                    const std::vector<unsigned int> new_map = {n_pt};
+
+                    cell_container[cell_id].push_back(all_cells[o]);
+                    qpoints_container[cell_id].push_back(new_qpoint);
+                    maps_container[cell_id].push_back(new_map);
```

Comment:
You could avoid creating the temporary objects if you use `emplace_back` instead.

Reply:
I tried to use emplace_back, but I couldn't understand how to remove completely the temporary vector. This wouldn't compile and I don't understand where I am wrong:
`qpoints_container[cell_id].emplace_back( std::vector<Point<dim0>> ( all_qpoints[o][j]) );`